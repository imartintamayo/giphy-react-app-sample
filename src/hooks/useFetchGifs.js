import { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        const getImages = async () => {
            const gifs = await getGifs(category);
            setState({
                data: gifs,
                loading: false
            });
        };
        getImages();
    }, [category]);

    return state;
}