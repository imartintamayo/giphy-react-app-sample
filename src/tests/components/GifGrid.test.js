import { shallow } from "enzyme";
import '@testing-library/jest-dom';

import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe('test <GifGrid />', () => {
    const category = 'some category';

    test('should be displayed correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should display items when useFetchGifs has resolved data', () => {
        const gifs = [{
            id: 'someId',
            url: 'someUrl',
            title: 'someTitle'
        }, {
            id: 'someId2',
            url: 'someUrl2',
            title: 'someTitle2'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
})
