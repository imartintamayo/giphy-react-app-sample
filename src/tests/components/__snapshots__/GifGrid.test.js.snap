// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <GifGrid /> should be displayed correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__rotateInDownLeft"
  >
    some category
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`test <GifGrid /> should display items when useFetchGifs has resolved data 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__rotateInDownLeft"
  >
    some category
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="someId"
      key="someId"
      title="someTitle"
      url="someUrl"
    />
    <GifGridItem
      id="someId2"
      key="someId2"
      title="someTitle2"
      url="someUrl2"
    />
  </div>
</Fragment>
`;
